src/mito2/src/worker/handle_write.rs

```diff
@@ -75,13 +76,21 @@ impl<S: LogStore> RegionWorkerLoop<S> {
                 .start_timer();
             let mut wal_writer = self.wal.writer();
             for region_ctx in region_ctxs.values_mut() {
+                if let WalOptions::Noop = &region_ctx.version().options.wal_options {
```

Comment:
Do we need to add a unit test that the wal is disabled?

Reply:
Added a sqlness case for that https://github.com/GreptimeTeam/greptimedb/pull/5740/commits/2c067d85a27a9076a7224e15c1e747e2aa957d0f#diff-0d620b4ddaeef2e408d31f800ca10cf11d8c64aa9884772cdaba5cf46d20c911